src/main/java/net/mcreator/ui/modgui/ArmorGUI.java

```diff
@@ -949,13 +954,13 @@ private void updateArmorTexturePreview() {
 		bootsModelPartR.setSelectedItem(armor.bootsModelPartR);
 
 		helmetCollapsiblePanel.toggleVisibility(
-				helmetModel.getSelectedItem() != defaultModel || !helmetSpecialInfo.getText().isEmpty());
+				helmetModel.getSelectedItem() != defaultModel || helmetSpecialInformation.getSelectedProcedure() != null);
```

Comment:
In this code, getSelectedProcedure is never null as value is returned with empty name, so this will not work as you intended. You can test when opening armor editor, panels are always collapsed even if they shouldn't be.

You should also expend in case if there is fixed value present so this case should be handled too.

Reply:
I don't get it, if `getSelectedProcedure` is never null, why is it compared to null more than once in MCreator's code? (I searched with CMD+SHIFT+F)

Reply:
I recommend checking source code of selector. Regular selector returns null in case of no element, but typed selector returns either procedure or default value defined.

In either case, the bug is valid and happens and thus needs to be resolved

Reply:
I have tried to figure this out, but everything I think of doesn't work... @Defeatomizer could I have some help with this please? (if/when you have time)

Reply:
Klemen, did you mean Max should use this snippet here?
```java
		helmetCollapsiblePanel.toggleVisibility(
				helmetModel.getSelectedItem() != defaultModel || armor.helmetSpecialInformation.getName() != null
						|| !armor.helmetSpecialInformation.getFixedValue().equals(""));
		bodyCollapsiblePanel.toggleVisibility(
				bodyModel.getSelectedItem() != defaultModel || armor.bodySpecialInformation.getName() != null
						|| !armor.bodySpecialInformation.getFixedValue().equals(""));
		leggingsCollapsiblePanel.toggleVisibility(
				leggingsModel.getSelectedItem() != defaultModel || armor.leggingsSpecialInformation.getName() != null
						|| !armor.leggingsSpecialInformation.getFixedValue().equals(""));
		bootsCollapsiblePanel.toggleVisibility(
				bootsModel.getSelectedItem() != defaultModel || armor.bootsSpecialInformation.getName() != null
						|| !armor.bootsSpecialInformation.getFixedValue().equals(""));
```

Reply:
Thanks very much, I tested and it works!

Reply:
> Klemen, did you mean Max should use this snippet here?

Yep, exactly that